run.py

```diff
@@ -1,30 +1,77 @@
 import argparse
 
-from src.add_songs import create_db, add_track
+import config
+from src.ingest_data import run_ingest_data
+from src.clean_data import run_clean_data
+from src.generate_features import run_generate_features
+from src.create_db import run_create_db
 
-if __name__ == '__main__':
+if __name__ == "__main__":
 
-    # Add parsers for both creating a database and adding songs to it
-    parser = argparse.ArgumentParser(description="Create and/or add data to database")
+    # Add parsers for all functions in the pipeline
+    parser = argparse.ArgumentParser(description="Run model pipeline code")
     subparsers = parser.add_subparsers()
 
-    # Sub-parser for creating a database
-    sb_create = subparsers.add_parser("create_db", description="Create database")
-    sb_create.add_argument("--artist", default="Britney Spears", help="Artist of song to be added")
-    sb_create.add_argument("--title", default="Radar", help="Title of song to be added")
-    sb_create.add_argument("--album", default="Circus", help="Album of song being added.")
-    sb_create.add_argument("--engine_string", default='sqlite:///data/tracks.db',
-                           help="SQLAlchemy connection URI for database")
-    sb_create.set_defaults(func=create_db)
-
-    # Sub-parser for ingesting new data
-    sb_ingest = subparsers.add_parser("ingest", description="Add data to database")
-    sb_ingest.add_argument("--artist", default="Emancipator", help="Artist of song to be added")
-    sb_ingest.add_argument("--title", default="Minor Cause", help="Title of song to be added")
-    sb_ingest.add_argument("--album", default="Dusk to Dawn", help="Album of song being added")
-    sb_ingest.add_argument("--engine_string", default='sqlite:///data/tracks.db',
-                           help="SQLAlchemy connection URI for database")
-    sb_ingest.set_defaults(func=add_track)
+    # Sub-parser for ingesting data from source
+    sb_ingest = subparsers.add_parser(
+        "ingest", description="Ingest data from web source and upload raw file to S3."
+    )
+    sb_ingest.add_argument(
+        "--config", default=config.YAML_CONFIG, help="Location of configuration YAML"
+    )
+    sb_ingest.add_argument(
+        "--url", default=config.URL_LISTINGS, help="URL of source data"
```

Comment:
I like how you make your ingestion script configurable, i will say that having a config.py, config.env, AND .yml makes it confusing to figure out what will get pulled where.

Reply:
Good point, I agree.

My intention is for the `config.env` to store all environment variables, `config.py` to store configurations for running the code for the end-to-end application, and `modelconfig.yml` to store static configurations pertaining to only the model pipeline (e.g., static data file names, feature names, and other constants). The user would only need to update the `config.env` and `config.py` to be able to run the application. The `modelconfig.yml` is mostly for the developer, to document the model version for reproducibility, while (theoretically) iterating on development with trying different features, etc.

I am moving the S3 bucket and RDS database name variables to the `config.py` (I initially thought these wouldn't need to be changed by the user), and hopefully this explanation helps with understanding the purpose of each configuration file. I will make a note in the README too.